src/Symfony/Component/Form/Extension/Core/DataTransformer/DateIntervalToArrayTransformer.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\DataTransformer;
+
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+use Symfony\Component\Form\Exception\UnexpectedTypeException;
+
+/**
+ * Transforms between a normalized date interval and an interval string/array.
+ *
+ * @author Steffen Roßkamp <steffen.rosskamp@gimmickmedia.de>
+ */
+class DateIntervalToArrayTransformer implements DataTransformerInterface
+{
+    const YEARS = 'years';
+    const MONTHS = 'months';
+    const DAYS = 'days';
+    const HOURS = 'hours';
+    const MINUTES = 'minutes';
+    const SECONDS = 'seconds';
+    const INVERT = 'invert';
+
+    private static $availableFields = array(
+        self::YEARS => 'y',
+        self::MONTHS => 'm',
+        self::DAYS => 'd',
+        self::HOURS => 'h',
+        self::MINUTES => 'i',
+        self::SECONDS => 's',
+        self::INVERT => 'r',
+    );
+    private $fields;
+
+    /**
+     * @param string[] $fields The date fields
+     * @param bool     $pad    Whether to use padding
+     */
+    public function __construct(array $fields = null, $pad = false)
+    {
+        if (null === $fields) {
+            $fields = array('years', 'months', 'days', 'hours', 'minutes', 'seconds', 'invert');
```

Comment:
I would use an associative array here instead of plain array (because you often use `array_flip`, and `isset($array[$key])` is faster than `in_array($key, $array)`)